omnibus/package-scripts/agent-dmg/preinst

```diff
@@ -103,9 +103,25 @@ echo '# Deleting old datadog-agent files'
 rm -rf $INSTALL_DIR/agent || true
 rm -rf $INSTALL_DIR/checks.d || true
 
-echo "# Disabling the login launch of the app"
-sudo -u "$INSTALL_USER" osascript -e 'tell application "System Events" to delete every login item whose name is "Datadog Agent"'
-
+user_uid=$(echo "$UID")
+install_user_home=$(echo "$HOME")
+user_plist_file=${install_user_home}/Library/LaunchAgents/com.datadoghq.agent.plist
+gui_user_plist_file=${install_user_home}/Library/LaunchAgents/com.datadoghq.gui.plist
+if sudo -u "$INSTALL_USER" launchctl print "gui/$user_uid/com.datadoghq.agent" 1>/dev/null 2>/dev/null; then
+  echo "# Disabling the login launch of the app"
+  user_agent_version=$(datadog-agent version | grep -Eo '[6-7]\.[0-9]+\.[0-9]+' | head -1)
+  if ! printf '7.52.0\n%s\n' "$user_agent_version" | sort -V -C; then
```

Comment:
Is there a way for a user installing a version before 7.52 to use this new code?

I doubt it will ever be backported, couldn't you assume that the install script is now running 7.52+?